src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxNode.cs

```diff
@@ -115,15 +119,21 @@ private static void ComputeSyntaxTree(CSharpSyntaxNode node)
 
                 foreach (var n in nodes)
                 {
-                    var existingTree = Interlocked.CompareExchange(ref n._syntaxTree, tree, null);
+                    var existingTree =  n._syntaxTree;
                     if (existingTree != null)
                     {
-                        tree = existingTree;
+                        Debug.Assert(existingTree == tree, "how could this node belong to a different tree?");
+
+                        // yield the race
+                        break;
                     }
+                    n._syntaxTree = tree;
```

Comment:
Same answer as above. We had a lot of unnecessary expenses in this method for using CmpExch in this method and to assign exactly the same value. 
Trying to assign the same value is still a contention for rights to write.